### PR TITLE
[WIP] Move quantized::add_scalar and quantized::add_scalar_relu to ATen

### LINKED ISSUE #27097: [c10] c10 dispatch doesn't support tracing of scalars
Example:
```
import torch

def foo(x):
    return torch.ops.quantized.add_scalar(x, 3)

x = torch.quantize_per_tensor(torch.rand(3, 4), 0.01, 0, torch.qint8)

torch.jit.trace(foo, (x,))
```

Results in the following error
```
RuntimeError: unsupported input type: Scalar
```

That traces back to this line: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/register_c10_ops.cpp#L175

However, normal ATen ops support this functionality. Take `__ilshift__` as an example:

```
import torch

def foo(x):
    return x << 3

#x = torch.quantize_per_tensor(torch.rand(3, 4), 0.01, 0, torch.qint8)
x = torch.rand(3, 4)

traced = torch.jit.trace(foo, (x,))
print(traced.graph)
```
Which works and prints the following graph:
```
graph(%x : Float(3, 4)):
  %1 : int = prim::Constant[value=3]() # trace_sa2.py:4:0
  %2 : Float(3, 4) = aten::__lshift__(%x, %1) # trace_sa2.py:4:0
  return (%2)
```

We should have this functionality in tracing c10 ops

### PR DESCRIPTION
**Pull request by ngoldbaum**
This is a WIP attempt to fix #27097.

This compiles and I'm able to do `import torch` but when I access `pytorch.ops.quantized.add_scalar`, I get an error:

```
In [3]: torch.ops.quantized.add_scalar
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-4f39065448d7> in <module>
----> 1 torch.ops.quantized.add_scalar

~/pytorch/torch/_ops.py in __getattr__(self, op_name)
     59         # for overloads and raise an exception if there are more than one.
     60         qualified_op_name = '{}::{}'.format(self.name, op_name)
---> 61         op = torch._C._jit_get_operation(qualified_op_name)
     62         # let the script frontend know that op is identical to the builtin op
     63         # with qualified_op_name

RuntimeError: Found 2 overloads for operator quantized::add_scalar! Overloads are not supported from Python.
```

### CODE REVIEW

File: aten/src/ATen/core/op_registration/op_registration.h [hunk 1]
```diff
@@ -372,7 +372,11 @@ class CAFFE2_API RegisterOperators final {
       static_assert(!std::is_same<FuncType, KernelFunction::BoxedKernelFunction>::value, "Tried to register a stackbased (i.e. internal) kernel function using the public kernel<...>() API. Please either use the internal kernel(...) API or also implement the kernel function as defined by the public API.");
       static_assert(kernel_func != nullptr, "Kernel function cannot be nullptr");
 
-      at::globalATenDispatch().registerOp<FuncType>(dispatch_key, legacyATenSchema_->c_str(), kernel_func);
+      if (legacyATenSchema_.has_value()) {
+        at::globalATenDispatch().registerOp<FuncType>(dispatch_key, legacyATenSchema_->c_str(), kernel_func);
+      } else {
+        at::globalATenDispatch().registerOp<FuncType>(dispatch_key, c10::toString(schemaOrName_->right()).c_str(), kernel_func);          
+      }
```
Comment (ngoldbaum) [inline aten/src/ATen/core/op_registration/op_registration.h]: Is it OK to assume that `schemaOrName_.is_right()` is true or do I also have to add a check for the `schemaOrName_.is_left()` chase?

**Comment by ngoldbaum** [comment]
I seem to be able to avoid the error about duplicated overloads by deleting the `out` variants:

```
In [5]: torch.ops.quantized.add_scalar
Out[5]: <function torch._ops.quantized.PyCapsule.add_scalar>

In [6]: torch.ops.quantized.add_scalar_relu
Out[6]: <function torch._ops.quantized.PyCapsule.add_scalar_relu>
```

but then I guess there's no way to do the in-place operation without using e.g. `torch.ops.quantized.add_scalar_`.

**Comment by jamesr66a** [comment]
This is already implemented by this stack: https://github.com/pytorch/pytorch/pull/27258 I just haven’t pushed land on it yet

**Comment by ngoldbaum** [comment]
Fair enough, hope I’m not stepping on any toes. This was a fun a learning experience in the pytorch codebase anyway...

**Comment by ezyang** [comment]
@jamesr66a For future reference, Quansight is using the "assignees" field to figure out if high priority issues are being worked on or not. So assign things to yourself if you're working on it :)

**Pull request closed without merging**